src/org/ods/component/ScanWithAquaStage.groovy

```diff
@@ -0,0 +1,138 @@
+        String credentialsId = context.cdProject + "-"
+        if (!secretName) {
+            credentialsId = context.credentialsId
+            logger.info("No custom secretName was specified in the aqua ConfigMap, continuing with default " +
+                "credentialsId '" + credentialsId + "'...")
+        } else {
+            credentialsId += secretName
+        }
+        // reference of the image that was build in this pipeline run
+        String imageRef = getImageRef()
+        if (!imageRef) {
+            logger.info "Skipping as imageRef could not be retrieved. Possible reasons are:\n" +
+                "-> The aqua stage runs before the image build stage and hence no new image was created yet.\n" +
+                "-> The image build stage was not executed because the image was imported.\n" +
+                "-> The aqua stage and the image build stage have different values for 'resourceName' set."
+            return
+        }
+
+        String reportFile = "aqua-report.html"
+        int returnCode = scanViaCli(url, registry, imageRef, credentialsId, reportFile)
+        createBitbucketCodeInsightReport(url, registry, imageRef, returnCode)
+        archiveReport(!context.triggeredByOrchestrationPipeline, reportFile)
+    }
+
+    private String getImageRef() {
+        // take the image ref of the image that is being build in the image build stage
+        Map<String, String> buildInfo =
+            context.getBuildArtifactURIs().builds[options.resourceName] as Map<String, String>
+        if (buildInfo) {
+            String imageRef = buildInfo.image
+            return imageRef.substring(imageRef.indexOf("/") + 1)
+        }
+        return null
+    }
+
+    private int scanViaCli(String aquaUrl, String registry, String imageRef, String credentialsId, String reportFile) {
+        logger.startClocked(options.resourceName)
+        int returnCode = aqua.scanViaCli(aquaUrl, registry, imageRef, credentialsId, reportFile)
+        // see possible return codes at https://docs.aquasec.com/docs/scanner-cmd-scan#section-return-codes
+        switch (returnCode) {
+            case 0:
+                logger.info "Finished scan via Aqua CLI successfully!"
+                break
+            case 1:
+                steps.error "An error occurred in processing the scan request " +
+                    "(e.g. invalid command line options, image not pulled, operational error)."
+                break
+            case 4:
+                logger.info "The image scanned failed at least one of the Image Assurance Policies specified."
+                break
+            default:
+                steps.error "An unknown return code was returned: ${returnCode}"
+        }
+        logger.infoClocked(options.resourceName, "Aqua scan (via CLI)")
+        return returnCode
+    }
+
+    private createBitbucketCodeInsightReport(String aquaUrl, String registry, String imageRef, int returnCode) {
```

Comment:
@renedupont can you mention known report limitations (text length?) in a comment?